tests/drivers/python/v5_8/explicit_tx_multiple_run.py

```diff
@@ -0,0 +1,23 @@
+from typing import Set
+
+import neo4j
+
+MEMGRAPH_URL = "bolt://localhost:7687"
+
+driver = neo4j.GraphDatabase.driver(MEMGRAPH_URL, auth=("", ""))
+
+
+def fill_db(tx):
+    tx.run("UNWIND range(1, 2000) AS i CREATE (n:Node {id: i}) RETURN n")
+
+
+def run_queries(tx):
+    tx.run("match (n) return n;")  # A query that forces the result to have has_more=true
+    tx.run("match (n) return n limit 1;")  # Any query you can run
+
+
+with driver.session() as session:
+    tx = session.begin_transaction()
+    fill_db(tx)
+    run_queries(tx)
+    tx.commit()
```

Comment:
I can break this by running the following python script:
```
import neo4j

MEMGRAPH_URL = "bolt://localhost:7687"

driver = neo4j.GraphDatabase.driver(MEMGRAPH_URL)

def run_queries(tx):
    big_result = tx.run("match (n:Node) set n.id = 9999 return n;") #A query that forces the result to have has_more=true
    short_result = tx.run("match (n:Node) set n.id = 8888 return n;") #Any query you can run
    #print(short_result.values())
    print("\n\n\nBREAK\n\n\n")
    print(big_result.values())
    print(short_result.values())

with driver.session() as session:
    session.run("UNWIND RANGE(1,3000) AS idx CREATE (:Node{id:idx});")
    tx = session.begin_transaction()
    run_queries(tx)
    tx.commit()
```

Memgrapg trace:
```
[2023-11-29 15:56:44.630] [memgraph_log] [debug] [Run - memgraph] 'UNWIND RANGE(1,3000) AS idx CREATE (:Node{id:idx});'
PULL(0) -1
[2023-11-29 15:56:44.636] [memgraph_log] [debug] [Run - memgraph] 'UNWIND RANGE(3000,3100) AS idx CREATE (:Bla{id:idx});'
PULL(1) -1
[2023-11-29 15:56:44.636] [memgraph_log] [debug] [Run - memgraph] 'match (n:Node) set n.id = 9999 return n;'
RUN 0
PULL(2) -1
[2023-11-29 15:56:44.640] [memgraph_log] [debug] [Run - memgraph] 'match (n:Node) set n.id = 8888 return n;'
RUN 1
PULL(3) -1
PULL(4) 0
PULL(5) 0
PULL(6) -1
PULL(7) -1
```
The PULL message has two numbers; in the brackets an index that ++ on each call, at the end the qid (if not specified -1).

-1 just means do the last query. But we have interleaved PULLs, so I get a result that is a mix of 9999 and 8888 ids.

Reply:
Note, this breaks neo4j too. But I don't think we should support this in a way that can break our snapshot isolation.